tests/JustSaying.UnitTests/Messaging/Channels/Fakes/InMemoryServiceResolver.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Threading;
+using JustSaying.Fluent;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.Middleware.Backoff;
+using JustSaying.Messaging.Middleware.ErrorHandling;
+using JustSaying.Messaging.Middleware.MessageContext;
+using JustSaying.Messaging.Middleware.PostProcessing;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit.Abstractions;
+
+namespace JustSaying.UnitTests.Messaging.Channels.Fakes
+{
+    public class InMemoryServiceResolver : IServiceResolver, IHandlerResolver
```

Comment:
Doesn't this just duplicate the built-in implementation we have that wraps `IServiceResolver` already?

Reply:
Assuming you're referring to `ServiceProviderResolver`, it takes an already built `IServiceProvider` which would mean tests would need to create a service collection, register stuff, then create a ServiceProviderResolver from it. Totally doable, but this one streamlines it a bit by providing a configuration action for overrides, and registering some defaults. The ServiceProviderResolver ctor is also internal so can't be called from tests unless we change it.

We could create a static class with defaults so you could do something like 
```
var sp = new ServiceCollection().AddTestDefaults().BuildServiceProvider()
var resolver= new ServiceProviderResolver(sp)
```
which is pretty much what this does.